### PR TITLE
feat: accessibility fixes + e2e axe

### PR DESCRIPTION
**Pull request by ValentinNelu**
Before submitting a pull request, please make sure you have at least performed the following:

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) and [DEVELOPER.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/DEVELOPER.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated any applicable tests.
 - [ ] added/updated any applicable API documentation.
 - [ ] added/updated any applicable demos.

### CODE REVIEW

File: src/carousel/carousel.ts [hunk 1]
```diff
@@ -111,13 +111,25 @@ export class NgbSlide {
...
+		<button
+			class="carousel-control-next"
+			type="button"
+			(click)="arrowRight()"
+			*ngIf="showNavigationArrows"
+			aria-labelledby="button-span-next"
+		>
 			<span class="carousel-control-next-icon" aria-hidden="true"></span>
-			<span class="visually-hidden" i18n="@@ngb.carousel.next">Next</span>
+			<span class="visually-hidden" i18n="@@ngb.carousel.next" id="button-span-next">Next</span>
```
Comment (maxokorokov) [inline src/carousel/carousel.ts:132]: This will not be unique in case there are several carousels on the page

**Review by maxokorokov — CHANGES_REQUESTED** [review]
Just one comment about ids

File: src/carousel/carousel.ts (hunk 1 above)
Comment (ValentinNelu) [reply src/carousel/carousel.ts:132]: indeed, I am gonna use the same logic as in accordion for the generated id

Comment (ValentinNelu) [reply src/carousel/carousel.ts]: I made the change to add unique ids for the spans. Should we use that ids (that are generated for each carousel) also as id on the carousel itself as we do for the accordion with the input?

File: demo/src/app/components/components.e2e-spec.ts [hunk 2]
```diff
@@ -26,6 +27,30 @@ const COMPONENTS = [
...
+	nav: ['aria-allowed-attr', 'aria-required-children'],
+	offcanvas: [],
+	pagination: [],
+	popover: [],
+	progressbar: ['color-contrast'],
+	rating: ['aria-input-field-name'],
+	scrollspy: ['color-contrast'],
+	table: ['select-name'],
+	timepicker: ['color-contrast'], //because of the alert
+	toast: [],
```
Comment (maxokorokov) [inline demo/src/app/components/components.e2e-spec.ts:47]: I think you should add contrast to toasts as well

**Review by maxokorokov — APPROVED** [review]
Thanks a lot!

**Pull request merged**